### ISSUE TITLE
gopls: automated issue report (crash)

### ISSUE THREAD

**Issue by ori-saporta83** [golang/vscode-go#841]
gopls version: v0.5.1
gopls flags: 

ATTENTION: PLEASE PROVIDE THE DETAILS REQUESTED BELOW.

Describe what you observed.

Crash when trying to rename

<pre>panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x7dbd3a]

goroutine 27461 [running]:
golang.org/x/tools/internal/lsp/source.pathEnclosingInterval(0xc0003d3d80, 0x0, 0x0, 0x343, 0x343, 0x17c210ac77452f29, 0xa, 0x0, 0x0, 0x0, ...)
	  rename_check.go:850  0x5a
golang.org/x/tools/internal/lsp/source.(*renamer).checkStructField(0xc011e0a000, 0xc01bf352c0)
	  rename_check.go:393  0xaf
golang.org/x/tools/internal/lsp/source.(*renamer).check(0xc011e0a000, 0xefac20, 0xc01bf352c0)
	  rename_check.go:44  0x37a
golang.org/x/tools/internal/lsp/source.Rename(0xeeb440, 0xc01b5e21e0, 0xefc320, 0xc00a9f2f00, 0xeeb640, 0xc00009a2a0, 0x4030000000000000, 0x4014000000000000, 0xc002e3a302, 0xa, ...)
	  rename.go:126  0x605
golang.org/x/tools/internal/lsp.(*Server).rename(0xc0001b81e0, 0xeeb380, 0xc005b9c340, 0xc005cc2180, 0x0, 0x0, 0x0)
	  rename.go:20  0x186
golang.org/x/tools/internal/lsp.(*Server).Rename(0xc0001b81e0, 0xeeb380, 0xc005b9c340, 0xc005cc2180, 0xc005cc2180, 0x0, 0x0)
	  server_gen.go:148  0x49
golang.org/x/tools/internal/lsp/protocol.serverDispatch(0xeeb380, 0xc005b9c340, 0xefeea0, 0xc0001b81e0, 0xc01b5e2030, 0xeeb5c0, 0xc005b9c1c0, 0x0, 0xed89e0, 0xc002ae2000)
	  tsserver.go:399  0x14e2
golang.org/x/tools/internal/lsp/protocol.ServerHandler.func1(0xeeb380, 0xc005b9c340, 0xc01b5e2030, 0xeeb5c0, 0xc005b9c1c0, 0x19a1e6ebb9, 0x1345ce0)
	  protocol.go:63  0xc5
golang.org/x/tools/internal/lsp/lsprpc.handshaker.func1(0xeeb380, 0xc005b9c340, 0xc01b5e2030, 0xeeb5c0, 0xc005b9c1c0, 0x0, 0x0)
	  lsprpc.go:557  0x452
golang.org/x/tools/internal/jsonrpc2.MustReplyHandler.func1(0xeeb380, 0xc005b9c340, 0xc00f038400, 0xeeb5c0, 0xc005b9c1c0, 0xc00b5f0660, 0xe11ac0)
	  handler.go:35  0xcf
golang.org/x/tools/internal/jsonrpc2.AsyncHandler.func1.2(0xc00e16a8a0, 0xc012598510, 0xc0005d8140, 0xeeb380, 0xc005b9c340, 0xc00f038400, 0xeeb5c0, 0xc005b9c1c0)
	  handler.go:103  0x86
created by golang.org/x/tools/internal/jsonrpc2.AsyncHandler.func1
	  handler.go:100  0x173
[Error - 8:42:36 AM] </pre>

OPTIONAL: If you would like to share more information, you can attach your complete gopls logs.

NOTE: THESE MAY CONTAIN SENSITIVE INFORMATION ABOUT YOUR CODEBASE.
DO NOT SHARE LOGS IF YOU ARE WORKING IN A PRIVATE REPOSITORY.

**Comment by hyangah** [comment]
@ori-saporta83  Thanks for the report. Looks like a duplicate of golang/go#42170. 
Can you please check if the prerelease version of gopls fixes the issue?
Assuming your gopls is installed in `$GOPATH/bin`
```
$ cd ~
$ GO111MODULE=on go get golang.org/x/tools/gopls@v0.5.2-pre.1
```

**Comment by stamblerre** [comment]
There hasn't yet been a pre-release with this fix, but it will be fixed in gopls/v0.5.2-pre.2, which will be available tomorrow.

**Comment by hyangah** [comment]
The latest version of gopls is v0.5.5. The fix for this crash was released a while ago. Closing.
@ori-saporta83 If you encounter a crash with the new version, please open another issue. Thanks!
